lib/ansible/galaxy/token.py

```diff
@@ -35,6 +48,93 @@
 
 display = Display()
 
+AH_TOKEN_EXPIRES_N_MIN_EARLY = 1
+AH_AUTH_URL = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token"
+AH_JWKS_URI = "https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/certs"
+
+# https://www.rfc-editor.org/rfc/rfc7518#section-3.1
+REQUIRED_JWT_HEADER_KEYS = set(['alg', 'typ', 'kid'])
+REQUIRED_JWT_HEADER_VALUES = {
+    'alg': ['RS256'],
+    'typ': ['JWT'],
+}
+
+PublicNumbers = namedtuple("PublicNumbers", ["e", "n"])
+
+
+def validate_jwt_format(data):
+    if len(data.split('.')) != 3:
+        raise InvalidSignature(f"unsupported format: {data}")
+
+    if any(whitespace in data.rsplit('.')[0] for whitespace in string.whitespace):
+        raise InvalidSignature(f"JWS header and payload should not contain line breaks or whitespace: {data.split('.')[0]}")
```

Comment:
Why does this use `str.split()` but the conditional has `str.rsplit()`?

Reply:
it was a typo, fixed